features/feature_case/math/math-saturatef.cu

```diff
@@ -0,0 +1,36 @@
+// ====------ math-saturatef.cu---------- *- CUDA -* ----===////
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+// ===----------------------------------------------------------------------===//
+
+#include <cassert>
+#include <iostream>
+#include <limits>
+
+#include <cuda.h>
+
+__global__ void testSaturatef(char *const TestResults) {
+  TestResults[0] = (__saturatef(-0.1f) == 0.0f);
+  TestResults[1] = (__saturatef(1.1f) == 1.0f);
+  TestResults[2] = (__saturatef(0.5f) == 0.5f);
+  TestResults[3] =
+      (__saturatef(std::numeric_limits<float>::quiet_NaN()) == 0.0f);
+}
+
+int main() {
+  constexpr int NumberOfTests = 4;
+  char *TestResults;
+  cudaMallocManaged(&TestResults, NumberOfTests * sizeof(*TestResults));
+  testSaturatef<<<1, 1>>>(TestResults);
+  cudaDeviceSynchronize();
+  for (int i = 0; i < NumberOfTests; i++) {
+    if (TestResults[i] == 0) {
+      std::cout << "Test " << i << " failed" << std::endl;
+    }
+    assert(TestResults[i] != 0);
+  }
```

Comment:
the main() function should explictly return 0 on success, else fail.

Reply:
I've update the test to include explicit return statements.